File: backend/src/main/java/com/example/app/model/SysUser.java

package com.example.app.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDateTime;

/**
 * 统一用户表实体类
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "sys_user", indexes = {
    @Index(name = "idx_username", columnList = "username", unique = true)
})
public class SysUser {

    /**
     * 主键ID
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    /**
     * 用户名/登录账号
     */
    @Column(name = "username", nullable = false, length = 64)
    private String username;

    /**
     * 密码
     */
    @Column(name = "password", nullable = false, length = 255)
    private String password;

    /**
     * 真实姓名
     */
    @Column(name = "real_name", nullable = false, length = 64)
    private String realName;

    /**
     * 用户大类（STUDENT 学生 / EMPLOYEE 员工）
     */
    @Column(name = "user_type", nullable = false, length = 32)
    private String userType;

    /**
     * 员工角色
     */
    @Column(name = "employee_role", length = 64)
    private String employeeRole;

    /**
     * 所属部门/院系
     */
    @Column(name = "department", length = 128)
    private String department;

    /**
     * 联系方式
     */
    @Column(name = "contact_info", length = 64)
    private String contactInfo;

    /**
     * 账号状态
     */
    @Column(name = "status", length = 16)
    private String status = "正常";

    /**
     * 创建时间
     */
    @Column(name = "create_time")
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @Column(name = "update_time")
    private LocalDateTime updateTime;

    @PrePersist
    protected void onCreate() {
        createTime = LocalDateTime.now();
        updateTime = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updateTime = LocalDateTime.now();
    }
}

File: model/sys_user.py

from sqlalchemy import Column, Integer, String, DateTime
from sqlalchemy.sql import func
from core.database import Base


class SysUser(Base):
    __tablename__ = 'sys_user'
    
    id = Column(Integer, primary_key=True, autoincrement=True, comment='主键ID')
    username = Column(String(64), nullable=False, unique=True, comment='用户名/登录账号')
    password = Column(String(255), nullable=False, comment='密码')
    real_name = Column(String(64), nullable=False, comment='真实姓名')
    user_type = Column(String(32), nullable=False, comment='用户大类（STUDENT/EMPLOYEE）')
    employee_role = Column(String(64), nullable=True, comment='员工角色')
    department = Column(String(128), nullable=True, comment='所属部门/院系')
    contact_info = Column(String(64), nullable=True, comment='联系方式')
    status = Column(String(16), default='正常', comment='账号状态')
    create_time = Column(DateTime, server_default=func.now(), comment='创建时间')
    update_time = Column(DateTime, server_default=func.now(), onupdate=func.now(), comment='更新时间')
